ansible/roles/test/files/ptftests/vxlan-decap.py

```diff
@@ -0,0 +1,293 @@
+            print "  ToPC   = ", res_t
+            print
+            if self.vxlan_enabled:
+                self.assertTrue(res_v, "VxlanTest failed")
+            else:
+                self.assertFalse(res_v, "VxlanTest must be disabled")
+            self.assertTrue(res_f, "FromPC test failed")
+            self.assertTrue(res_t, "ToPC test failed")
+
+    def Vxlan(self, test):
+        rv = True
+        for n in self.net_ports:
+            for a in test['acc_ports']:
+                res = self.checkVxlan(a, n, test)
+                rv = rv and res
+
+        return rv
+
+    def FromPC(self, test):
+        rv = True
+        for n in self.net_ports:
+            for a in test['acc_ports']:
+                res = self.checkRegularFromPC(a, n, test)
+                rv = rv and res
+        return rv
+
+    def ToPC(self, test):
+        rv = True
+        for dst, ports in self.pc_info:
+            for a in test['acc_ports']:
+                res = self.checkRegularToPC(a, ports, dst, test)
+                rv = rv and res
+        return rv
+
+    def checkRegularToPC(self, acc_port, pc_ports, dst_ip, test):
+        rv = True
+        src_mac = self.ptf_mac_addrs['eth%d' % acc_port]
+        dst_mac = self.dut_mac
+        src_ip = test['vlan_ip_prefix'] % acc_port
+
+        packet = simple_tcp_packet(
+                         eth_dst=dst_mac,
+                         eth_src=src_mac,
+                         ip_src=src_ip,
+                         ip_dst=dst_ip,
+                       )
+        exp_packet = simple_tcp_packet(
+                         eth_dst=self.random_mac,
+                         eth_src=dst_mac,
+                         ip_src=src_ip,
+                         ip_dst=dst_ip,
+                         ip_ttl = 63,
+                       )
+
+        exp_packet = Mask(exp_packet)
+        exp_packet.set_do_not_care(0, 6*8)
+
+        for i in xrange(self.nr):
+            testutils.send_packet(self, acc_port, packet)
+        nr_rcvd = testutils.count_matched_packets_all_ports(self, exp_packet, pc_ports, timeout=0.1)
+        rv = rv and (nr_rcvd == self.nr)
+        return rv
+
+
+    def checkRegularFromPC(self, acc_port, net_port, test):
+        rv = True
+        src_mac = self.random_mac
+        dst_mac = self.dut_mac
+        src_ip = test['src_ip']
+        dst_ip = test['vlan_ip_prefix'] % acc_port
+
+        packet = simple_tcp_packet(
+                         eth_dst=dst_mac,
+                         eth_src=src_mac,
+                         ip_src=src_ip,
+                         ip_dst=dst_ip,
+                       )
+
+        exp_packet = simple_tcp_packet(
+                         eth_dst=self.ptf_mac_addrs['eth%d' % acc_port],
+                         eth_src=dst_mac,
+                         ip_src=src_ip,
+                         ip_dst=dst_ip,
+                         ip_ttl = 63,
+                       )
+
+        for i in xrange(self.nr):
```

Comment:
Do we need this `for` loop, since its executed only once.

Reply:
I'll add self.nr as a parameter to the test